ranger-server/src/main/java/com/flipkart/ranger/server/manager/RangerClientManager.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 Flipkart Internet Pvt. Ltd.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.ranger.server.manager;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.flipkart.ranger.client.Constants;
+import com.flipkart.ranger.client.zk.UnshardedRangerZKHubClient;
+import com.flipkart.ranger.core.model.Criteria;
+import com.flipkart.ranger.server.config.RangerConfiguration;
+import com.flipkart.ranger.zookeeper.serde.ZkNodeDataDeserializer;
+import com.google.common.base.Preconditions;
+import io.dropwizard.lifecycle.Managed;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.RetryForever;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Slf4j
+@Singleton
+@Getter
+public class RangerClientManager<T, C extends Criteria<T>, D extends ZkNodeDataDeserializer<T>> implements Managed {
+
+    private final RangerConfiguration rangerConfiguration;
+
+    private final UnshardedRangerZKHubClient<T, C> zkHubClient;
```

Comment:
You might want to rethink this. Ideally this should be abstract here, so that the server can, if it wants, add some in-memory cached decorator to the hub